pylint/lint.py

```diff
@@ -946,12 +946,20 @@ def register_checker(self, checker):
 
         :raises ValueError: If the priority of the checker is invalid.
         """
-        if checker.name in self._checkers:
-            # TODO: Raise if classes are the same
-            for duplicate in self._checkers[checker.name]:
-                msg = 'A checker called {} has already been registered ({}).'
-                msg = msg.format(checker.name, duplicate.__class__)
-                warnings.warn(msg)
+        existing_checker_types = set(
+            type(existing_checker)
+            for name_checkers in self._checkers.values()
+            for existing_checker in name_checkers
+        )
```

Comment:
Could cache this between calls, but not for simplicity.

Reply:
To avoid having to create an entire set we could do something like the following?

```
existing_checker_types = (type(existing_checker) for name_checkers in self._checkers.values() for existing_checker in name_checkers)
if any(existing_type == checker_type for existing type in existing_checker_types):
   ...
```

Reply:
I thought it read a little smoother. The set has better worst case performance with around 1000 classes too. Little investigation here https://gist.github.com/fffergal/cbdf43bb29d52be46da4a34ddbf0f0c7.